Source/Blazorise/Utilities/AsyncCountdownTimer.cs

```diff
@@ -0,0 +1,155 @@
+﻿#region Using directives
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+#endregion
+
+namespace Blazorise.Utilities;
+
+/// <summary>
+/// Represents a countdown timer that can be extended, paused, and resumed, and allows for actions to be taken on each tick and upon completion.
+/// </summary>
+public class AsyncCountdownTimer : IDisposable
```

Comment:
I think I wanted something similar to this before, could this be superior to what we have in other places? Snackbar? Carousel?

Reply:
I like it more because it more modern approach with the PeriodicTimer instead of an older Timer. This one also have native support for async code.

Reply:
Yes, But I rememember I tried using this more modern approach before in Carousel with no success, but I probably couldn't implement a proper Timer? It's whether you think it would be appropriate to update the other implementations